animations-dbus/animations-dbus-client.c

```diff
@@ -0,0 +1,544 @@
+   * com.endlessm.Libanimation:/com/endlessm/Libanimation/ConnectionManager */
+  animations_dbus_connection_manager_proxy_new (priv->connection,
+                                                G_DBUS_PROXY_FLAGS_NONE,
+                                                "com.endlessm.Libanimation",
+                                                "/com/endlessm/Libanimation/ConnectionManager",
+                                                g_task_get_cancellable (task),
+                                                on_created_connection_manager_proxy,
+                                                task);
+}
+
+static gboolean
+animations_dbus_client_init_finish (GAsyncInitable  *initable,
+                                    GAsyncResult    *result,
+                                    GError         **error)
+{
+  GTask *task = G_TASK (result);
+  g_return_val_if_fail (g_task_is_valid (task, initable), FALSE);
+
+  return g_task_propagate_boolean (task, error);
+}
+
+static void
+animations_dbus_client_init_async (GAsyncInitable      *initable,
+                                   int                  io_priority G_GNUC_UNUSED,
+                                   GCancellable        *cancellable,
+                                   GAsyncReadyCallback  callback,
+                                   gpointer             user_data)
+{
+  AnimationsDbusClient *client = ANIMATIONS_DBUS_CLIENT (initable);
+  AnimationsDbusClientPrivate *priv = animations_dbus_client_get_instance_private (client);
+  g_autoptr(GTask) task = g_task_new (initable, cancellable, callback, user_data);
+
+  g_task_set_task_data (task, client, NULL);
+
+  if (priv->animation_manager_proxy != NULL)
+    {
+      g_task_return_boolean (task, TRUE);
+      return;
+    }
+
+  g_bus_get (G_BUS_TYPE_SESSION,
+             cancellable,
+             on_got_session_bus,
+             g_steal_pointer (&task));
+}
+
+static void
+animations_dbus_client_async_initable_interface_init (GAsyncInitableIface *iface)
+{
+  iface->init_async = animations_dbus_client_init_async;
+  iface->init_finish = animations_dbus_client_init_finish;
+}
+
+static void
+animations_dbus_client_dispose (GObject *object)
+{
+  AnimationsDbusClient *client = ANIMATIONS_DBUS_CLIENT (object);
+  AnimationsDbusClientPrivate *priv = animations_dbus_client_get_instance_private (client);
+
+  g_clear_object (&priv->animation_manager_proxy);
+  g_clear_object (&priv->connection_manager_proxy);
+  g_clear_object (&priv->connection);
+
+  G_OBJECT_CLASS (animations_dbus_client_parent_class)->dispose (object);
+}
+
+static void
+animations_dbus_client_init (AnimationsDbusClient *client G_GNUC_UNUSED)
+{
+}
+
+static void
+animations_dbus_client_class_init (AnimationsDbusClientClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->dispose = animations_dbus_client_dispose;
+}
+
+void
+animations_dbus_client_new_async (GCancellable                      *cancellable,
```

Comment:
I would have clients pass the dbus connection here

Reply:
See my thoughts on this above - we could potentially do that, but I'm curious to know if owning a separate connection is bad practice.